### PR TITLE
Search - make term/prefix/wildcard/regex query parsing more lenient

### LINKED ISSUE #63893: Update semantics or definition of `case_insensitive` in term queries
PR https://github.com/elastic/elasticsearch/pull/61596 added a new `case_insensitive` boolean flag to term queries. This is a boolean flag that can only be set to `true`, which is rather unusual.

After talking to @markharwood it turns out the semantics of this flag is as follows:
- not set: uses the field's definition (which may or not include lowercase normalization)
- set to `true`: case insensitive
- set to `false`: not supported as we can't enforce case sensitivity if the field has been normalized with a lowercase normalizer.

From a Clients perspective, this makes this flag very much a snowflake. We can represent it as an optional boolean, but users will still have to be aware that this boolean, when set, can only be set to `true`. If the value for this flag comes from an application variable like a checkbox in a search dialog, users will have to add some logic to only set it if `true` and not if `false`.

So I suggest that we either change the semantics of this flag (and update the docs accordingly) or change its definition.

Change semantics:
* `true`: case insensitive
* `false`: use the field's definition

Change definition: rename it to `case_sensitivity` with values
* `field_default` - use the field's setting
* `insensitive` - case insensitive
* and this leaves a door open to a future addition of `sensitive` if that make sense.

Changing the semantics can be enough if we're sure that `sensitive` doesn't (and will never) make sense, and will just be about updating the doc and removing some validation checks.

The enumeration will still require some custom logic in applications, but won't be surprising as it will be a type conversion from a boolean (checkbox) to an enum.

/cc @markharwood @jimczi

### PR DESCRIPTION
**Pull request by markharwood**
Remove errors when case_insensitive flag set to false
Closes #63893

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/query/PrefixQueryBuilder.java [hunk 1]
```diff
@@ -109,9 +109,6 @@ public String value() {
     }
 
     public PrefixQueryBuilder caseInsensitive(boolean caseInsensitive) {
-        if (caseInsensitive == false) {
-            throw new IllegalArgumentException("The case insensitive setting cannot be set to false.");
-        }
```
Comment (mayya-sharipova) [inline server/src/main/java/org/elasticsearch/index/query/PrefixQueryBuilder.java:114]: What's the plan here?  To allow clients to set `caseInsensitive` to `false`?

Comment (markharwood) [reply server/src/main/java/org/elasticsearch/index/query/PrefixQueryBuilder.java:114]: Yep. Makes some of the clients easier to write

**Comment by markharwood** [comment]
test this please

**Comment by markharwood** [comment]
test this please

**Review by jimczi — APPROVED** [review]
LGTM

**Pull request merged**